tests/unit/test_useragent.py

```diff
@@ -65,20 +65,22 @@ def test_basic_user_agent_string():
         Config(retries={'mode': 'legacy'}, user_agent_appid='fooapp')
     )
 
-    actual = ua.to_string()
-    expected = (
-        f'Botocore/{botocore_version} '
-        'md/awscrt#Unknown '
-        'ua/2.0 '
-        'os/linux#1.2.3-foo '
-        'md/arch#x86_64 '
-        'lang/python#3.8.20 '
-        'md/pyimpl#Dpython '
-        'exec-env/AWS_Lambda_python3.8 '
-        'cfg/retry-mode#legacy '
-        'app/fooapp'
-    )
-    assert actual == expected
+    actual = ua.to_string().split(' ')
+    expected_in_exact_order = [
+        f'Botocore/{botocore_version}',
+        'md/awscrt#Unknown',
+        'ua/2.0',
+        'os/linux#1.2.3-foo',
+        'md/arch#x86_64',
+        'lang/python#3.8.20',
+        'md/pyimpl#Dpython',
+        'exec-env/AWS_Lambda_python3.8',
+        'cfg/retry-mode#legacy',
+        'app/fooapp',
+    ]
+
+    indices = [actual.index(el) for el in expected_in_exact_order]
+    assert indices == list(sorted(indices)), 'Elements were found out of order'
```

Comment:
Can we potentially do something like this on the tests rather than rewriting them to try to work around injected ordering?

```python
@mock.patch.object(botocore.useragent, 'modify_components', noop_modify)
```

Reply:
I reverted these changes and instead added `@mock.patch.object` to pass a function that doesn't modify components.